p2p/proto/common.proto

```diff
@@ -1,88 +1,41 @@
 syntax = "proto3";
 
-message FieldElement
-{
+message Felt252 {
     bytes elements = 1;
 }
 
-message PeerID
-{
-    bytes id = 1;
+message Hash {
+    bytes elements = 1;
 }
 
-message BlockHeader
-{
-    FieldElement parent_block_hash = 1;
-    uint64 block_number = 2;
-    FieldElement global_state_root = 3;
-    FieldElement sequencer_address = 4;
-    uint64 block_timestamp = 5;
-
-    uint32 transaction_count = 6;
-    FieldElement transaction_commitment = 7;
-
-    uint32 event_count = 8;
-    FieldElement event_commitment = 9;
-
-    uint32 protocol_version = 10;
-
+message Address {
+    bytes elements = 1;
 }
 
-message InvokeTransaction
-{
-    FieldElement contract_address = 1;
-    FieldElement entry_point_selector = 2;
-    repeated FieldElement calldata = 3;
-    repeated FieldElement signature = 4;
-    FieldElement max_fee = 5;
-    FieldElement nonce = 6;
-    FieldElement version = 7;
+message PeerID {
+    bytes id = 1;
 }
 
-
-message DeclareTransaction
-{
-    ContractClass contract_class = 1;
-    FieldElement sender_address = 2;
-    FieldElement max_fee = 3;
-    repeated FieldElement signature = 4;
-    FieldElement nonce = 5;
-    FieldElement version = 6;
+message ChainID {
+    bytes id = 1;
 }
 
-message ContractClass
-{
-    message EntryPoint {
-        FieldElement selector = 1;
-        FieldElement offset = 2;
-    }
+message BlockID {
+    Hash hash = 1;
+    uint64 height = 2;
```

Comment:
If we're not using `oneof` 3 possible variants here (hash, height, hash&height) I suggest we at least make height optional, otherwise `BlockID {h,0}`
[is ambiguous](https://protobuf.dev/programming-guides/proto3/#default), we don't know if it refers to:
- genesis block with some hash `h`:
```
BlockID {
    hash: h,
    height: 0,
}
```
- a block with some hash `h`, where height is not specified:
```
BlockID {
    hash: h
}
```


```suggestion
    Hash hash = 1;
    optional uint64 height = 2;
```

Reply:
This will require protobuf >= 3.15 btw

Reply:
I agree this should be oneof

Reply:
ok, so oneof then :+1:

Reply:
Note that I'm not sure anymore we need to query by hash, if not, then block id will be just the block number (and the hash will be in the header)

Reply:
even better tbh